### PR TITLE
TCA setup

### PR DESCRIPTION
**Pull request by pedroseruca**
Imported TCA module.
Added a root view under TCA guidelines, with a simple dummy button which navigates forward in the stack.

This is an experiment to understand how TCA and SwiftUI link up.

### CODE REVIEW

File: MinderaPeople/Screens/RootView.swift [hunk 1]
```diff
@@ -0,0 +1,97 @@
+//
+//  RootView.swift
```
Comment (p4checo) [inline MinderaPeople/Screens/RootView.swift:2]: we could probably get rid of these, by creating a `IDETemplateMacros.plist` inside the `xcshareddata` of our project/workspace. I think we only have a project for now, but might be a good idea to create a workspace.

The file's contents could be:
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<!--
    Text macros reference:
    https://help.apple.com/xcode/mac/9.0/index.html?localePath=en.lproj#/dev91a7a31fc
    https://help.apple.com/xcode/mac/9.0/index.html?localePath=en.lproj#/dev7fe737ce0

    Examples:
    https://oleb.net/blog/2017/07/xcode-9-text-macros/
-->
<plist version="1.0">
<dict>
    <key>FILEHEADER</key>
    <string></string>
</dict>
</plist>
```

File: MinderaPeople/Screens/RootView.swift [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+//  MinderaPeople
+//
+//  Created by Mindera on 06/10/2022.
+//
+
+import ComposableArchitecture
+import SwiftUI
+
+struct AppState: Equatable {
```
Comment (p4checo) [inline MinderaPeople/Screens/RootView.swift:11]: I think that we should use the latest TCA idiom which uses `ReducerProtocol` and a new dependency management system. 

You can take a look at the relevant (free) episodes which explain it nicely:
- https://www.pointfree.co/episodes/ep201-reducer-protocol-the-problem
- https://www.pointfree.co/episodes/ep208-reducer-protocol-in-practice

File: MinderaPeople/Screens/RootView.swift [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    case logInButtonTapped
+    case homePageDismiss
+}
+
+let appReducer = Reducer<AppState, AppAction, AppEnvironment> { state, action, _ in
+    switch action {
+    case .logInButtonTapped:
+        state.isShowingHomePage = true
+        return .none
+    case .homePageDismiss:
```
Comment (p4checo) [inline MinderaPeople/Screens/RootView.swift:28]: missing newline between multiline cases 🚓

File: MinderaPeople/MinderaPeopleApp.swift [hunk 4]
```diff
@@ -5,13 +5,23 @@
...
 struct MinderaPeopleApp: App {
+    private let state = AppState()
+    private let environment = AppEnvironment()
+    
     var body: some Scene {
         WindowGroup {
-            ContentView()
+            RootView(
+                store: .init(
+                    initialState: state,
```
Comment (p4checo) [inline MinderaPeople/MinderaPeopleApp.swift:20]: this seems to fit a single line (120 chars) 💅🏼

File: MinderaPeople/Screens/RootView.swift [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+                            .padding()
+                            .background {
+                                Capsule()
+                                    .foregroundColor(.yellow)
+                            }
+                    }
+                }
+                .padding()
+                .navigationDestination(
+                    isPresented: viewStore.binding(
```
Comment (p4checo) [inline MinderaPeople/Screens/RootView.swift:63]: shouldn't `.binding(` be on the next line? 🤔

File: MinderaPeople/Screens/RootView.swift [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+                VStack {
+                    Spacer()
+                    VStack {
+                        Image(systemName: "globe")
+                            .imageScale(.large)
+                            .foregroundColor(.accentColor)
+                        Text("Hello, world!")
+                    }
+                    Spacer()
+
```
Comment (p4checo) [inline MinderaPeople/Screens/RootView.swift:48]: spacing between views is not consistent 💅🏼

**Review by p4checo — CHANGES_REQUESTED** [review]
Nice one! 🙌🏼 Some suggestions related with using latest TCA and code style (I'm assuming we will follow https://github.com/Mindera/swift-style-guide 😇)

**Review by asos-tiagoveloso — APPROVED** [review]
Looks good to me, awesome kick start.
But I do agree with @p4checo's suggestions

File: MinderaPeople/MinderaPeopleApp.swift (hunk 4 above)
Comment (pedroseruca) [reply MinderaPeople/MinderaPeopleApp.swift]: done

File: MinderaPeople/Screens/RootView.swift (hunk 5 above)
Comment (pedroseruca) [reply MinderaPeople/Screens/RootView.swift]: done

File: MinderaPeople/Screens/RootView.swift (hunk 1 above)
Comment (pedroseruca) [reply MinderaPeople/Screens/RootView.swift]: done

File: MinderaPeople/Screens/RootView.swift [hunk 7]
```diff
@@ -0,0 +1,97 @@
+//
```
Comment (p4checo) [inline MinderaPeople/Screens/RootView.swift:1]: sadly we can't remove this `//` and newline automatically, so can you please remove it manually?

(same for other files)

File: MinderaPeople/MinderaPeopleApp.swift [hunk 8]
```diff
@@ -1,17 +1,15 @@
...
 import SwiftUI
 
 @main
 struct MinderaPeopleApp: App {
+    private let feature = RootFeature()
+
     var body: some Scene {
         WindowGroup {
-            ContentView()
+            RootView(store: .init(initialState: .init(), reducer: feature.body))
```
Comment (p4checo) [inline MinderaPeople/MinderaPeopleApp.swift:12]: do we need to pass in the `.body`? (same case in the previews)

File: MinderaPeople/Screens/RootView.swift [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+
+import ComposableArchitecture
+import SwiftUI
+
+struct RootFeature: ReducerProtocol {
```
Comment (p4checo) [inline MinderaPeople/Screens/RootView.swift:6]: any particular reason to call it `RootFeature` and not just `Root`?

**Review by p4checo — APPROVED** [review]
Great stuff! Just some minor tweaks here and there 🤓

File: MinderaPeople/Screens/RootView.swift (hunk 7 above)
Comment (pedroseruca) [reply MinderaPeople/Screens/RootView.swift]: done

File: MinderaPeople/MinderaPeopleApp.swift (hunk 8 above)
Comment (pedroseruca) [reply MinderaPeople/MinderaPeopleApp.swift]: done

File: MinderaPeople/Screens/RootView.swift (hunk 9 above)
Comment (pedroseruca) [reply MinderaPeople/Screens/RootView.swift]: change name to RootReducer, conforming to some example from isoWords project

**Pull request merged**